client/src/partials/cash/modals/transfer.modal.js

```diff
@@ -0,0 +1,113 @@
+angular.module('bhima.controllers')
+.controller('CashTransferModalController', CashTransferModalController);
+
+CashTransferModalController.$inject = [
+  '$uibModalInstance', '$q', 'SessionService', 'CurrencyService', 'VoucherService', 'CashboxService', 'AccountService', 'cashBox', 'uuid'
+];
+
+/**
+ * @module cash/modals/CashTransferModalController
+ *
+ * @description This controller is responsible transfering money between auxillary cash and a virement account
+*/
+
+function CashTransferModalController(ModalInstance, $q, sessionService, currencyService, voucherService, cashBoxService, accountService, cashBox, uuid) {
+  var vm = this; 
+
+  /** Attaching service to the scope **/
+  vm.cashBox = cashBox;
+  vm.currencyService = currencyService; 
+  vm.cashBoxService = cashBoxService;
+
+  /** check if the cashbox id is provided**/
+  vm.cashBoxIdMissed = !cashBox.id;
+
+  /**
+  * @function submit
+  * @param {boolean} invalid it is true if the form is valid
+  * @description for submitting a dialog content
+  **/
+  function submit (invalid){
+    if (invalid) { return; }
+    var voucher = processVoucher();
+    var voucher_item = processVoucherItem(voucher.uuid);
+    var record = { voucher : voucher, voucher_item : voucher_item };
+    voucherService.create(record)      
+    .then(function (res){
+      vm.submited = true; //FIX ME, I need a goo idea
+    })
+    .catch(function (err){
+      vm.fail = true; //FIX ME, I need a goo idea
```

Comment:
`vm.fail` will not be reset if a user makes a successful submission.  Maybe use one variable for both the success and error states, and another for the "submitted" states?  For example:
```js
// default to no successful submission on module init.
vm.success = false;

function submit(invalid) {
  if (invalid) { return; }
  // ... code ...
  voucherService.create(record)
  .then(function (res) {
    vm.success = true;
  });
}
```
Then, in your view, you could have:
```html
<p ng-show="VoucherModalForm.$submitted && Ctrl.success">
  Yay!  Success!
</p>
<p ng-show="VoucherModalForm.$submitted && !Ctrl.success">
  Oh No!  Failure!
</p>
```
In this case, the form is handling the `submitted` state, and you are only handling the success/error states.  You could also try to manually manage the `submitted` state independent of the form.